### PR TITLE
Fix MacOS installation instructions.

### PR DESCRIPTION
**Pull request by daggerok**
Update instructions to get binary from GitHub release on MacOS: `kind-Darwin-amd64 -> kind-darwin-amd64`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,7 +37,15 @@ generally recommended for CI usage in particular.
 To install, download the binary for your platform from "Assets" and place this
 into your `$PATH`:
 
-On Mac & Linux:
+On Mac:
+
+```console
+curl -Lo ./kind https://github.com/kubernetes-sigs/kind/releases/download/v0.5.1/kind-darwin-amd64
```
Comment (aojea) [inline README.md]: I think that $(uname) returns `darwin` on mac, no need to differentiate

Comment (BenTheElder) [reply README.md]: it returns `Darwin`, however, https://github.com/kubernetes-sigs/kind/releases/download/v0.5.1/kind-Darwin-amd64 works just fine, and we unconditionally save it as `./kind`

**Comment by BenTheElder** [comment]
thanks but, these should already work, and don't need separating 🙃

**Comment by daggerok** [comment]
Hello, it doesn't worked for me when I have tried it according to documentation:
uname returns Darwin with capital letter, but your release page contains in lowercase

File: README.md (hunk 1 above)
Comment (daggerok) [reply README.md]: I got 404

**Comment by BenTheElder** [comment]
```
[bentheelder@bentheelder-macbookpro:~/junk·2019-11-15T19:03:08-0800]
$ curl -Lo ./kind https://github.com/kubernetes-sigs/kind/releases/download/v0.5.1/kind-$(uname)-amd64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   608    0   608    0     0   2063      0 --:--:-- --:--:-- --:--:--  2068
100 36.4M  100 36.4M    0     0  2592k      0  0:00:14  0:00:14 --:--:-- 5966k

[bentheelder@bentheelder-macbookpro:~/junk·2019-11-15T19:03:23-0800]
$ stat ./kind
16777220 33203388 -rw-r--r-- 1 bentheelder primarygroup 0 38215872 "Nov 15 19:03:24 2019" "Nov 15 19:03:23 2019" "Nov 15 19:03:23 2019" "Nov 15 19:03:09 2019" 4096 75848 0 ./kind
```

**Comment by daggerok** [comment]
Hello,

@BenTheElder, you are right..
Seems like I have used `curl -O ...` to save files with ther original name but it doesn't worked wit that command.

My apologies

---
Regards

**Comment by BenTheElder** [comment]
ah ok great! I was trying to find how this could break 😅

**Pull request closed without merging**